wasp-ai/src/client/pages/UserPage.jsx

```diff
@@ -100,8 +124,54 @@ function UserProjectsTable({ projects }) {
                 </Link>
               </td>
             </tr>
-          ))}
+          ))
+        ) : (
+          <tr className="bg-white border-t">
+            <td colSpan={5} className="text-center py-4">
+              you have not generated any apps yet.
+            </td>
+          </tr>
+        )}
       </tbody>
     </table>
   );
-}
\ No newline at end of file
+}
+
+function DeleteUserModal({ isOpen, setIsOpen, deleteUser, user }) {
+  async function deleteUserHandler() {
+    await deleteUser({ userId: user.id });
+    logout();
+  }
+
+  return (
+    <MyDialog
+      isOpen={isOpen}
+      onClose={() => setIsOpen(false)}
+      title={
+        <div className="flex items-center gap-2">
+          <PiSealWarningDuotone arie-hidden="true" /> Are You Sure You Want to Delete Your Account?
+        </div>
+      }
+    >
+      <div className="mt-10 space-y-10">
+        <p className="px-8 text-base leading-relaxed text-center text-gray-500">
+          You will lose access to your current projects and data.
```

Comment:
Are they just loosing access, or are we actually deleting their data? If I am correct, the latter? If so, I would communicate that clearly for two reasons:
1. So it is clear to them they can't later restore this access.
2. So it is clear to them that we will not keep their data on our side.

So we could say smth like "We will delete all your current projects and data and you won't be able to restore it."

Reply:
No, I cascaded the SocialLogin entity, not their projects